src/extra/stream/client.ts

```diff
@@ -83,6 +86,84 @@ export class StreamClient extends EventDispatcher implements Stream {
     }
   }
 
+  /**
+   * Get the available livestream resolutions for a camera component. If the
+   * stream client cannot find any available resolutions, an empty list will be
+   * returned.
+   *
+   * @param resourceName - The name of a camera component.
+   * @returns A list of available resolutions for livestreaming.
+   */
+  async getOptions(resourceName: string): Promise<Resolution[]> {
+    const fetchOptions = async (name: string): Promise<Resolution[]> => {
+      const request = new GetStreamOptionsRequest({ name });
+      this.options.requestLogger?.(request);
+      try {
+        const response = await this.client.getStreamOptions(request);
+        return response.resolutions;
+      } catch {
+        return [];
+      }
+    };
+
+    const trackName = getValidSDPTrackName(resourceName);
+    let resolutions = await fetchOptions(trackName);
+    if (resolutions.length > 0) {
+      return resolutions;
+    }
+    // Second attempt with resource name
+    resolutions = await fetchOptions(resourceName);
+    return resolutions;
+  }
+
+  /**
+   * Set the livestream options for a camera component. This will change the
+   * resolution of the stream to the specified width and height.
+   *
+   * @param name - The name of a camera component.
+   * @param width - The width of the resolution.
+   * @param height - The height of the resolution.
+   */
+  async setOptions(name: string, width: number, height: number) {
+    const request = new SetStreamOptionsRequest({
+      name: getValidSDPTrackName(name),
+      resolution: {
+        width,
+        height,
+      },
+    });
+    this.options.requestLogger?.(request);
+    try {
+      await this.client.setStreamOptions(request);
+    } catch {
+      // Try again with just the resource name
```

Comment:
why do we try again with just the resource name? if getValidSDPTrackName(name) is false then we are just falling back to using an unformatted name?

Reply:
Not sure... Was initially thrown off by this but this is the behavior from the other stream client endpionts so I copied it. If we want I can do a follow up PR to test this out and remove across the stream client if we dont need it.

Reply:
Looks like the flutter sdk only tries the sdp track name.
https://github.com/viamrobotics/viam-flutter-sdk/blob/main/lib/src/media/stream/client.dart#L88-L97

Reply:
Unfortunately I don't know enough on whether one way is better than another here but I think stick with consistency here on the other stream clients?